### ISSUE TITLE
OpenAPI spec parameters are mapped incorrectly

### ISSUE THREAD

**Issue by fardjad** [seriousme/fastify-openapi-glue#447]
Hello,

First off, I'd like to thank you for your amazing work. I think I've found an issue with OpenAPI spec parameters mapping. Here's a minimal example:

Given the following OpenAPI spec document:

```yaml
# api.oas.yml

openapi: 3.0.3
info:
  title: Test
  version: 0.0.0
paths:
  /test:
    get:
      parameters:
        - in: query
          name: param
          schema:
            type: array
      responses:
        "200":
          description: OK
          content:
            application/json:
              schema:
                type: object
                properties: {}
```

And the following JS file:

```javascript
// index.mjs

import { fileURLToPath } from "node:url";
import path from "node:path";
import Fastify from "fastify"; // v4.11.0
import openapiGlue from "fastify-openapi-glue"; // v4.1.1

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const fastify = Fastify();

// for debugging purposes
fastify.addHook("preValidation", (request, reply, done) => {
  console.log(request.routeSchema);
  done();
});

const options = {
  specification: path.join(__dirname, "./api.oas.yml"),
  service: {
    async getTest(request, reply) {
      return {};
    },
  },
};
fastify.register(openapiGlue, options);
await fastify.listen({ port: 3000 });
```

When I send the following request to the server:

```bash
curl http://localhost:3000/test
```

I receive the following response:

```
{"statusCode":400,"error":"Bad Request","message":"querystring must be array"}
```

And here's the `routeSchema` object that is logged earlier:

```
{
  querystring: { type: 'array' },
  response: { '200': { type: 'object', properties: {} } },
  [Symbol(fastify.schemas.visited)]: true
}
```

As you can see, when there is only a single query parameter defined in an operation, it generates an invalid querystring schema.

Interestingly enough, when I define an extra query parameter, it works as expected and produces the following `querystring` schema:

```
{
  type: 'object',
  properties: { param: { type: 'array' }, param2: { type: 'array' } }
}
```

Can you please have a look?

**Comment by fardjad** [comment]
Oh sorry, I removed the item type while I was trying to make the example as minimal as possible. Even with the following valid schema:

```
# api.oas.yml

openapi: 3.0.3
info:
  title: Test
  version: 0.0.0
paths:
  /test:
    get:
      parameters:
        - in: query
          name: param
          schema:
            type: array
            items:
              type: string
      responses:
        "200":
          description: OK
          content:
            application/json:
              schema:
                type: object
                properties: {}
```

it still maps the parameters definition to the following:

```
{
  ...
  querystring: { type: 'array', items: { type: 'string' } }
}
```

Also, I'm afraid the problem is not related to the out of box querystring parsing limitations of Fastify. Since the parameter is optional, it doesn't need to parse anything when it's not specified. The problem is, when we only have one querystring parameter, it wrongly defines the querystring schema as an array (instead of an object) in the route validation schema.

**Comment by seriousme** [comment]
Hi,

the complexity in the check unfortunately comes from complexity in the spec:
https://spec.openapis.org/oas/v3.1.0#fixed-fields-9

Adding an explicit `explode:false` to the schema should fix the issue for now.
I will check the details tomorrow as we probably need a patch with associated tests here.

Kind regards,
Hans

**Comment by seriousme** [comment]
Published on NPM as 4.1.2

Thanks for reporting the issue!
Hans
